client/consul_template_test.go

```diff
@@ -563,11 +564,12 @@ func TestTaskTemplateManager_Unblock_Consul(t *testing.T) {
 
 func TestTaskTemplateManager_Unblock_Vault(t *testing.T) {
 	t.Parallel()
+	require := require.New(t)
 	// Make a template that will render based on a key in Vault
-	vaultPath := "secret/password"
+	vaultPath := "secret/data/password"
 	key := "password"
 	content := "barbaz"
-	embedded := fmt.Sprintf(`{{with secret "%s"}}{{.Data.%s}}{{end}}`, vaultPath, key)
+	embedded := fmt.Sprintf(`{{with secret "%s"}}{{.Data.data.%s}}{{end}}`, vaultPath, key)
```

Comment:
Yikes, the format is backward incompatible? Seems like this is going to bite basically ever Vault+Nomad user. Anything we can do?

Reply:
It is a little complicated but it is backwards compatible. When you mount the secret backend in vault you get to pick v1 or v2. But since our unit tests use the dev server, they are mounting v2 by default. So our tests break b/c they were using v1 syntax. But users have to upgrade from v1 -> v2 if not using a dev server.